waku/node/waku_node.nim

```diff
@@ -182,6 +184,22 @@ proc connectToNodes*(node: WakuNode, nodes: seq[RemotePeerInfo] | seq[string], s
   # NOTE Connects to the node without a give protocol, which automatically creates streams for relay
   await peer_manager.connectToNodes(node.peerManager, nodes, source=source)
 
+## Waku Sync
+
+proc mountWakuSync*(node: WakuNode): Result[void, string] =
```

Comment:
wondering if sync could be part of store protocol?

Reply:
Yes but I'm sure there are some use cases where operators might want Sync but not Store.

Reply:
Guess in that case you can instantiate the store protocol but with a flag specifying to not serve other nodes and just sync?

No strong opinion, but having sync as another protocol seems odd to me.

Reply:
Sync will probably have it's own DB and needs to be searchable in the DHT but it could be part of Store for sure.

Code wise it seams best to separate both Store and Sync since they don't have the same responsibility.

I have no strong opinion either both could work. I'll try to think about the pros/cons of both.

Reply:
Although in terms of configuration and operation Store and Sync will function as a unit (and therefore a "single protocol") on the wire the one is concerned with querying a key-value store and the other with reconciliation of a set of keys. This requires different request-responses so would have to be a separate libp2p protocol ID. For that reason it makes sense to me to have some code separation at lowest layer, even if at a higher layer there will be integration.

Reply:
> Although in terms of configuration and operation Store and Sync will function as a unit (and therefore a "single protocol") on the wire the one is concerned with querying a key-value store and the other with reconciliation of a set of keys. This requires different request-responses so would have to be a separate libp2p protocol ID. For that reason it makes sense to me to have some code separation at lowest layer, even if at a higher layer there will be integration.

I was thinking that the App layer would be where we tie Store and SYNC.